### PR TITLE
Day 10

### PR DESCRIPTION
**Pull request by MirekSh**
image
 Test Card Numbers:
- Visa Card: 49927398716
- AMEX: 378282246310005
- Discover: 6011111111111117
- MasterCard: 5555555555554444

### CODE REVIEW

File: solutions/09/script.js [hunk 1]
```diff
@@ -1,30 +1,63 @@
...
-    var sum = 0;
-    for (var i = 0; i < val.length; i++) {
-        var intVal = parseInt(val.substr(i, 1));
-        if (i % 2 == 0) {
-            intVal *= 2;
-            if (intVal > 9) {
-                intVal = 1 + (intVal % 10);
-            }
+function whatCard(value) {
+    const icon = document.querySelector('.icon');
```
Comment (wokayme) [inline solutions/09/script.js]: const icon = document.querySelector('.icon');
better get it once, not every time when you run function.

File: solutions/09/script.js [hunk 2]
```diff
@@ -1,30 +1,63 @@
-function validateCardNumber(number) {
-    var regex = new RegExp("^[0-9]{16}$");
-    if (!regex.test(number))
-        return false;
+function validateCardNumber(value) {
+    // Accept only digits, dashes or spaces
+	if (/[^0-9-\s]+/.test(value)) return false;
 
-    return luhnCheck(number);
+	// The Luhn Algorithm. It's so pretty.
```
Comment (wokayme) [inline solutions/09/script.js]: Instead of commend, make separate function

File: solutions/09/script.js [hunk 3]
```diff
@@ -1,30 +1,63 @@
...
-    console.log("validate -> myCard.value", myCard.value)
-    validateCardNumber(myCard.value);
+function formValidation(e)
+{
+    e.preventDefault();
+    const myCard = document.querySelector('#cardNumber');
+    const val = myCard.value;
+    const err = document.querySelector("#err");
+    if (!validateCardNumber(val)) {
+        err.innerHTML="Wrong Card Number";
```
Comment (wokayme) [inline solutions/09/script.js]: textContent will be faster ;)

File: solutions/09/script.js [hunk 4]
```diff
@@ -1,30 +1,63 @@
...
-const myCard = document.querySelector('#card-number-field');
 
-function validate() {
-    console.log("validate -> myCard.value", myCard.value)
-    validateCardNumber(myCard.value);
+function formValidation(e)
+{
+    e.preventDefault();
+    const myCard = document.querySelector('#cardNumber');
+    const val = myCard.value;
```
Comment (wokayme) [inline solutions/09/script.js]: The same, you don't need to get buttons every time when you click "validate form"

File: solutions/09/script.js (hunk 1 above)
Comment (wokayme) [reply solutions/09/script.js]: What about old class?

Comment (MirekSh) [reply solutions/09/script.js]: Good point with const icon. What do you mean by 'old class'

File: solutions/09/script.js (hunk 2 above)
Comment (MirekSh) [reply solutions/09/script.js]: Done

File: solutions/09/script.js (hunk 3 above)
Comment (MirekSh) [reply solutions/09/script.js]: 👍

File: solutions/09/script.js (hunk 4 above)
Comment (MirekSh) [reply solutions/09/script.js]: Done :)

File: solutions/09-11/script.js [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+function cvvValidation()
+{
+    !(/^[0-9]{3,4}$/.test(cvv.value)) ?
+        errCvv.textContent="Wrong CVV value" :
+        errCvv.textContent = "";
+}
+
+function ownerValidation() {
+    !(/^[a-zA-Z\s]*$/.test(owner.value)) ?
+        errOwner.textContent="Use only letters and spaces" :
```
Comment (wokayme) [inline solutions/09-11/script.js:76]: I think it doesn't work

File: solutions/09-11/script.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+		nCheck += nDigit;
+		bEven = !bEven;
+	}
+	return (nCheck % 10) == 0;
+}
+
+// Card recognition function - first char is reserved for card type
+function whatCard(value) {
+    switch(value[0]) {
+        case '3': {
```
Comment (wokayme) [inline solutions/09-11/script.js]: I don't know what means that cases, this number can make problems, maybe better make map object:
```

const typeCards = {
visa: 'visa',
mastercard: 'mastercard;
}
```

File: solutions/09-11/script.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+        err.textContent="Wrong Card Number";
+    } else {
+        err.textContent = "";
+        whatCard(val);
+    }
+}
+
+function cvvValidation()
+{
+    !(/^[0-9]{3,4}$/.test(cvv.value)) ?
```
Comment (wokayme) [inline solutions/09-11/script.js]: try avoid using regest

File: solutions/09-11/creditCard.html [hunk 8]
```diff
@@ -3,32 +3,36 @@
...
-                    <div class="expirtion-date" id="expiration-date">
+                    <div class="expiration-date" id="expiration-date">
                         <label>Expiration Date</label>
-                        <input type="text" class="form-control" id="exp-date">
+                        <input type="text" class="form-control" id="exp-date" placeholder="MM/YY">
+                        <span id="errDate"></span>
                     </div>
                     <div class="cvv" id="cvv-number">
                         <label>CVV</label>
                         <input type="text" class="form-control" id="cvv">
```
Comment (wokayme) [inline solutions/09-11/creditCard.html]: Type number, with min and max value can be help full

File: solutions/09-11/script.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+function ownerValidation() {
+    !(/^[a-zA-Z\s]*$/.test(owner.value)) ?
+        errOwner.textContent="Use only letters and spaces" :
+        errOwner.textContent = "";
+}
+
+function dateValidation() {
+    const date = expDate.value.split('/');
```
Comment (wokayme) [inline solutions/09-11/script.js]: IMO better is transform user date for Date object and later make opperation

File: solutions/09-11/script.js [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    const currentDate = new Date();
+    const year = currentDate.getFullYear();
+    const yearInDate = String(year).slice(-2);
+    if (expDate.value.length === 5) {
+        (date[0] < 1 || date[0] > 12) || date[1] < yearInDate ? errDate.textContent = "Wrong date" :
+        errDate.textContent = "";
+    }
+}
+
+
```
Comment (wokayme) [inline solutions/09-11/script.js:96]: Try work a bit on composition, Firstly create two type of function. First only for validate data for example:
`isCsvValid(csv)`
and another type of function which will `display` changes in document

File: solutions/09/script.js (hunk 1 above)
Comment (wokayme) [reply solutions/09/script.js]:
```
icon.className = "";
            icon.classList.add('icon');
            icon.classList.add('fa');
```

Comment (MirekSh) [reply solutions/09/script.js]: Changed to `icon.className = "icon fa";`

File: solutions/09-11/script.js (hunk 5 above)
Comment (MirekSh) [reply solutions/09-11/script.js:76]: image
image
I've tested it :P

File: solutions/09-11/script.js (hunk 6 above)
Comment (MirekSh) [reply solutions/09-11/script.js]: Good point, cardTypes object added

File: solutions/09-11/script.js (hunk 7 above)
Comment (MirekSh) [reply solutions/09-11/script.js]: 👍

File: solutions/09-11/creditCard.html (hunk 8 above)
Comment (MirekSh) [reply solutions/09-11/creditCard.html]: I forgot about it, it was little tricky to display span if input:invalid :)

File: solutions/09-11/script.js (hunk 9 above)
Comment (MirekSh) [reply solutions/09-11/script.js]: Date object have problems with format MM/YY and it resolves it to 20/03 2001 :/

File: solutions/09-11/script.js (hunk 10 above)
Comment (MirekSh) [reply solutions/09-11/script.js:96]: I've created the formatValidation func and have used it in dateValidation

**Pull request merged**